example-app/src/main/java/com/adyen/checkout/example/ui/giftcard/GiftCardActivity.kt

```diff
@@ -0,0 +1,157 @@
+
+        setupReloadButton()
+
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                launch { giftCardViewModel.giftCardComponentDataFlow.collect(::setupGiftCardView) }
+                launch { giftCardViewModel.giftCardViewStateFlow.collect(::onGiftCardViewState) }
+                launch { giftCardViewModel.events.collect(::onGiftCardEvent) }
+            }
+        }
+    }
+
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+
+        val data = intent.data
+        if (data != null && data.toString().startsWith(RedirectComponent.REDIRECT_RESULT_SCHEME)) {
+            giftCardComponent?.handleIntent(intent)
+        }
+    }
+
+    private fun onGiftCardViewState(giftCardViewState: GiftCardViewState) {
+        when (giftCardViewState) {
+            GiftCardViewState.Loading -> {
+                // We are hiding the CardView here to display our own loading state. If you leave the view visible
+                // the built in loading state will be shown.
+                binding.progressIndicator.isVisible = true
+                binding.giftCardContainer.isVisible = false
+                binding.errorView.isVisible = false
+            }
+            GiftCardViewState.ShowComponent -> {
+                binding.giftCardContainer.isVisible = true
+                binding.progressIndicator.isVisible = false
+                binding.errorView.isVisible = false
+            }
+            GiftCardViewState.Error -> {
+                binding.errorView.isVisible = true
+                binding.progressIndicator.isVisible = false
+                binding.giftCardContainer.isVisible = false
+            }
+        }
+    }
+
+    private fun setupGiftCardView(giftCardComponentData: GiftCardComponentData) {
+        val giftCardComponent = GiftCardComponent.PROVIDER.get(
+            activity = this,
+            paymentMethod = giftCardComponentData.paymentMethod,
+            configuration = checkoutConfigurationProvider.getGiftCardConfiguration(),
+            callback = giftCardComponentData.callback,
+        )
+
+        this.giftCardComponent = giftCardComponent
+
+        binding.giftCardView.attach(giftCardComponent, this)
+    }
+
+    private fun setupReloadButton() {
+        binding.loadNewGiftCard.setOnClickListener {
+            giftCardViewModel.reloadComponentWithOrder()
+        }
+    }
+
+    private fun reloadGiftCardWithOrder(giftCardComponentData: GiftCardComponentData, orderRequest: OrderRequest) {
+        val giftCardComponent = GiftCardComponent.PROVIDER.get(
+            activity = this,
+            paymentMethod = giftCardComponentData.paymentMethod,
+            configuration = checkoutConfigurationProvider.getGiftCardConfiguration(),
+            callback = giftCardComponentData.callback,
+            order = orderRequest,
+            key = KEY_SECONDARY_GIFT_CARD_COMPONENT
```

Comment:
Should we use a key for the first component as well?

Reply:
I think we should only introduce the key if strictly necessary. We can think later on since we're gonna work on example integration further.